### PR TITLE
Properly handle "invalid" symbols during Level 1 snapshot operations

### PR DESCRIPTION
**Pull request by mrut2pac**
Added logic to handle invalid symbol snapshot queries to fail fast and with a proper exception rather than a timeout.

### CODE REVIEW

**Comment by mathpaquette** [comment]
nice @mrut2pac this is good. Did you check if you can add the request also?

**Comment by mathpaquette** [comment]
@mrut2pac by the way, I was about to raise a PR to replace task cancelation with TimeoutIQFeedException instead.

File: src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs [hunk 1]
```diff
@@ -59,12 +61,20 @@ void Level1ClientOnFundamental(FundamentalMessage fundamentalMessage)
...
 
+            void Level1ClientOnSymbolNotFound(SymbolNotFoundMessage symbolNotFoundMessage)
+            {
+                if (symbolNotFoundMessage.Symbol == symbol)
+                    res.TrySetException(new SymbolNotFoundIQFeedException(symbol));
+            }
+
             _level1MessageHandler.Fundamental += Level1ClientOnFundamental;
+            _level1MessageHandler.SymbolNotFound += Level1ClientOnSymbolNotFound;
             ReqWatch(symbol);
```
Comment (mathpaquette) [inline src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs:72]: what do you think if we return the request as a string from ReqWatch like
```
private string ReqWatch(string symbol)
        {
            var request = _level1RequestFormatter.ReqWatch(symbol);
            _socketClient.Send(request);
            return request;
        }
```

that way, we can reuse the request in the exception to make sure we have something consistent for all exceptions

same for ReqUnwatch

**Review by mathpaquette — CHANGES_REQUESTED** [review]
thats great stuff. adding current request in the Exception is a nice to have.

File: src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs (hunk 1 above)
Comment (mathpaquette) [reply src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs:72]: @mrut2pac could you please also add a quick integration test. this is quickly achieve.

Comment (mrut2pac) [reply src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs:72]: That's what I was doing in the beginning, but then I realized the delegate is set before calling ReqWatch, therefore it need to be either handled in ReqWatch or we need to break it down. Let me see what's the best way of doing it.
We are not handling any errors for ReqUnwatch, do we need it?
Adding a test is a good idea. I will do it.

Comment (mathpaquette) [reply src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs:72]: ahh yes you're right. Than, maybe just call the requestFormatter in the delegate...

Comment (mrut2pac) [reply src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs]: split the request creation and sending to get more elegant solution

Comment (mathpaquette) [reply src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs]: perfect!

**Review by mathpaquette — APPROVED** [review]

**Pull request merged**